### PR TITLE
Add removeIfHasKey

### LINKED ISSUE #21235: Add map:removeIfHasKey
**Description:**
Please refer https://github.com/ballerina-platform/ballerina-spec/issues/381

**Affected Versions:**
ballerina 1.1.0

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

Fixes #21235

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+
+import static org.ballerinalang.jvm.MapUtils.checkIsMapOnlyOperation;
+import static org.ballerinalang.jvm.MapUtils.checkValidFieldForRecord;
+
+/**
+ * Extern function to remove element from the map if key exists.
+ * ballerina.model.map:removeIfHasKey(string)
```
Comment (anupama-pathirage) [inline langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java]: Shall we add `@since` annotation for this?

Comment (KavinduZoysa) [reply langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java]: Added.

File: langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "lang.map", functionName = "removeIfHasKey",
+        args = {@Argument(name = "m", type = TypeKind.MAP), @Argument(name = "k", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.ANY)},
+        isPublic = true
+)
+public class RemoveIfHasKey {
+
+    public static Object removeIfHasKey(Strand strand, MapValue<?, ?> m, String k) {
+        String op = "removeIfHasKey()";
```
Comment (pubudu91) [inline langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java]: Shall we move this to a constant?

File: langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        checkIsMapOnlyOperation(m.getType(), op);
+        checkValidFieldForRecord(m, k, op);
+        try {
+            return m.remove(k);
+        } catch (org.ballerinalang.jvm.util.exceptions.BLangFreezeException e) {
+            throw BallerinaErrors.createError(e.getMessage(),
+                    "Failed to remove element from map: " + e.getDetail());
```
Comment (pubudu91) [inline langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java]: Need to verify whether this is consistent with how freeze panics are handled in other places. (i.e., the error message and its format)

File: langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java (hunk 2 above)
Comment (KavinduZoysa) [reply langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java]: Added.

File: langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java (hunk 3 above)
Comment (KavinduZoysa) [reply langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/RemoveIfHasKey.java]: Yes, this is consistent with other places where the freeze panics are handled.

**Pull request merged**